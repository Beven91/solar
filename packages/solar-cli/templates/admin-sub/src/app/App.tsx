
import React from 'react';
import { store } from '$projectName$-provider';
import { CrashProvider } from 'solar-pc';
import Router from './router';

const Provider = store.Provider;

export default function SubAdminApp() {
  return (
    <div style={{ height: '100%' }} className="$projectName$-sub-main">
      <Provider store={store.store}>
        <CrashProvider>
          <Router />
        </CrashProvider>
      </Provider>
    </div>
  );
}
