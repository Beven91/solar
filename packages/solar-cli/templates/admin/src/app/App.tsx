
import './index.less';
import React from 'react';
import { ConfigProvider } from 'antd';
import zhCN from 'antd/lib/locale-provider/zh_CN';
import { store } from '$projectName$-provider';
import { CrashProvider } from 'solar-pc';
import Router from './router';

const Provider = store.Provider;

export default function FluxyAdminApp() {
  return (
    <div className="top-container $projectName$-root">
      <Provider store={store.store}>
        <ConfigProvider locale={zhCN}>
          <CrashProvider>
            <Router />
          </CrashProvider>
        </ConfigProvider>
      </Provider>
    </div>
  );
}
