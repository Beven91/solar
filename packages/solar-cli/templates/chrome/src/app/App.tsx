import './index.less';
import React from 'react';
import { ConfigProvider, message, notification } from 'antd';
import { CrashProvider } from 'solar-pc';
import Router from './router';

export default class App extends React.Component {
  popupRoot: HTMLDivElement

  constructor(props: any) {
    super(props);
    this.popupRoot = document.createElement('div');
    this.popupRoot.className = 'chrome';
    document.body.appendChild(this.popupRoot);
    // 配置message container
    message.config({
      getContainer: () => this.popupRoot,
    });
    notification.config({
      getContainer: () => this.popupRoot,
    });
  }

  /**
   * 配置antd用于添加css作用域
   */
  get configProvider() {
    return {
      getPopupContainer: () => this.popupRoot,
    };
  }

  render() {
    return (
      <div className="chrome">
        <ConfigProvider getPopupContainer {...this.configProvider}>
          <CrashProvider>
            <Router />
          </CrashProvider>
        </ConfigProvider>
      </div>
    );
  }
}