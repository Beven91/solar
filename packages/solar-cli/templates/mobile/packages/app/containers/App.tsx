import './css/index.scss';
import React from 'react';
import { CrashProvider } from 'solar-mobile';
import Router from './router';

export default class App extends React.Component {
  render() {
    return (
      <div className="top-container">
        <CrashProvider>
          <Router />
        </CrashProvider>
      </div>
    );
  }
}