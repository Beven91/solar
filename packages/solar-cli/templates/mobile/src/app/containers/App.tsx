import './css/index.scss';
import React from 'react';
import { CrashProvider } from 'solar-mobile';
import Router from './router';

export default class App extends React.Component {
  render() {
    return (
      <div className="app-root-view">
        <CrashProvider>
          <Router />
        </CrashProvider>
      </div>
    );
  }
}