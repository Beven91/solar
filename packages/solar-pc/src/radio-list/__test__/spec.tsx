import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../demo/index.basic';

describe('radio-list', () => {
  it('renders correctly', () => {
    const tree = mount(
      <App />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
