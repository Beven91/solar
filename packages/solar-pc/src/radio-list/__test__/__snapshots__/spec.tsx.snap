// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio-list renders correctly 1`] = `
<RadioListDemo>
  <RadioList
    data={
      Array [
        Object {
          "label": "美国",
          "value": 1,
        },
        Object {
          "label": "英国",
          "value": 2,
        },
      ]
    }
    direction="vertical"
    labelName="label"
    onChange={[Function]}
    value=""
    valueName="value"
  >
    <ForwardRef
      className=" radio-list  vertical"
      data={
        Array [
          Object {
            "label": "美国",
            "value": 1,
          },
          Object {
            "label": "英国",
            "value": 2,
          },
        ]
      }
      labelName="label"
      onChange={[Function]}
      value=""
      valueName="value"
    >
      <div
        className="ant-radio-group ant-radio-group-outline  radio-list  vertical"
      >
        <Radio
          className="radio-list-item"
          key="1"
          type="radio"
          value={1}
        >
          <label
            className="ant-radio-wrapper radio-list-item"
          >
            <Checkbox
              checked={false}
              className=""
              defaultChecked={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              prefixCls="ant-radio"
              style={Object {}}
              type="radio"
              value={1}
            >
              <span
                className="ant-radio"
                style={Object {}}
              >
                <input
                  checked={false}
                  className="ant-radio-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value={1}
                />
                <span
                  className="ant-radio-inner"
                />
              </span>
            </Checkbox>
            <span>
              美国
            </span>
          </label>
        </Radio>
        <Radio
          className="radio-list-item"
          key="2"
          type="radio"
          value={2}
        >
          <label
            className="ant-radio-wrapper radio-list-item"
          >
            <Checkbox
              checked={false}
              className=""
              defaultChecked={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              prefixCls="ant-radio"
              style={Object {}}
              type="radio"
              value={2}
            >
              <span
                className="ant-radio"
                style={Object {}}
              >
                <input
                  checked={false}
                  className="ant-radio-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value={2}
                />
                <span
                  className="ant-radio-inner"
                />
              </span>
            </Checkbox>
            <span>
              英国
            </span>
          </label>
        </Radio>
      </div>
    </ForwardRef>
  </RadioList>
</RadioListDemo>
`;
